qa/standalone/scrub/osd-recovery-scrub.sh

```diff
@@ -131,10 +132,121 @@ function TEST_recovery_scrub_1() {
     return 0
 }
 
+##
+# a modified version of wait_for_scrub(), which terminates if the Primary
+# of the to-be-scrubbed PG changes
+#
+# Given the *last_scrub*, wait for scrub to happen on **pgid**. It
+# will fail if scrub does not complete within $TIMEOUT seconds. The
+# repair is complete whenever the **get_last_scrub_stamp** function
+# reports a timestamp different from the one given in argument.
+#
+# @param pgid the id of the PG
+# @param the primary OSD when started
+# @param last_scrub timestamp of the last scrub for *pgid*
+# @return 0 on success, 1 on error
+#
+function wait_for_scrub_mod() {
+    local pgid=$1
+    local orig_primary=$2
+    local last_scrub="$3"
+    local sname=${4:-last_scrub_stamp}
+
+    for ((i=0; i < $TIMEOUT; i++)); do
+        sleep 0.2
+        if test "$(get_last_scrub_stamp $pgid $sname)" '>' "$last_scrub" ; then
+            return 0
+        fi
+        sleep 1
+        # are we still the primary?
```

Comment:
Why is the primary changing?  If there is a legitimate primary change, the scrub request could be lost.  Too bad we don't re-request the scrub instead of just stopping waiting.

Reply:
Not sure about the "why" part. But if that happens - the test should handle that correctly.
We can possibly add a separate test that restarts the test in that specific case. But first - we should
get the existing tests to never fail.

Reply:
I'm worried that this is covering another problem in the test or in the code.  It's worth identifying why this behaviour is happening in the first place.

Reply:
Trying (as in fixing the other bugs reported on the scrub code), but do not want to block this PR until then.

Reply:
I'm ok with it if @dzafman is.